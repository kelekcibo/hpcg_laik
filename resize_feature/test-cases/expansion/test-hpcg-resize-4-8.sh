#!/bin/sh

dir="./test-cases/expansion"

# TEST CASE: 4 initial processes, 4 new joining processes after calling laik_allow_world_resize()
./launcher/tcp2run -n 4 -s 4 ./../bin/xhpcg

# Move result-file into .out file to be able to compare the results
mv ./hpcg*.txt ${dir}/test-hpcg-resize-4-8.out 

# Compare output with expected output
if cmp ${dir}/test-hpcg-resize-4-8.out "${dir}/test-hpcg-resize-4-8.expected"; 
    then echo "Test Case: Result is as expected";
else
    echo "Test Case: Result is not as expected";
fi