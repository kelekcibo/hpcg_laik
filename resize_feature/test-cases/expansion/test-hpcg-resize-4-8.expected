REPARTIONING: Old world size [4] New world size [8]
REPARTITIONG: Call [0] Scaled Residual [3.77205e-20]
WARNING: PERFORMING UNPRECONDITIONED ITERATIONS
Call [0] Number of Iterations [11] Scaled Residual [1.04927e-14]
WARNING: PERFORMING UNPRECONDITIONED ITERATIONS
Call [1] Number of Iterations [11] Scaled Residual [1.04927e-14]
Call [0] Number of Iterations [2] Scaled Residual [9.09606e-17]
Call [1] Number of Iterations [2] Scaled Residual [9.09606e-17]
Departure from symmetry (scaled) for SpMV abs(x'*A*y - y'*A*x) = 0
Departure from symmetry (scaled) for MG abs(x'*Minv*y - y'*Minv*x) = 5.40014e-08
SpMV call [0] Residual [0]
SpMV call [1] Residual [0]
Number of CG sets: 1
Call [0] Scaled Residual [1.18667e-22]
